helm: allow to disable local auth in teleport-cluster chart (#13827)
gravitational/teleport: docs/pages/setup/helm-reference/teleport-cluster.mdx

`kubeClusterName` sets the name used for the Kubernetes cluster. This name will be shown to Teleport users connecting to the cluster.

## `authentication`

> ### `authentication.type`

Reviewer: I think all field-specific headings are H2s.
```suggestion
## `authentication.type`
```

Author: I checked and we use h3 in the reference for nested values, I'll correct all other titles

| Type | Default value | Required? | `teleport.yaml` equivalent | Can be used in `custom` mode? |
| - | - | - | - | - |
| `string` | `local` | Yes | `auth_service.authentication.type` | ❌ |

> `authentication.type` controls the authentication scheme used by Teleport. Possible values are `local` and `github` for OSS, plus `oidc`, `saml` for Enterprise.

Reviewer 2: ```suggestion
`authentication.type` controls the authentication scheme used by Teleport. Possible values are `local` and `github` for OSS, plus `oidc` and `saml` for Enterprise.
```

> `authentication.type` controls the authentication scheme used by Teleport. Possible values are `local` and `github` for OSS, plus `oidc`, `saml` for Enterprise.

Reviewer 2: ```suggestion
`authentication.type` controls the authentication scheme used by Teleport. Possible values are `local` and `github` for OSS, plus `oidc` and `saml` for Enterprise.
```

## `authentication.localAuth`

| Type | Default value | Required? | `teleport.yaml` equivalent | Can be used in `custom` mode? |
| - | - | - | - | - |
| `bool` | `true` | No | `auth_service.authentication.local_auth` | ❌ |

`authentication.localAuth` controls whether local authentication is enabled.

> When disabled, users can only login through non-local auth like `saml`, `oidc` or `github`.

Reviewer: ```suggestion
When disabled, users can only log in through authentication connectors like `saml`, `oidc` or `github`.
```
"login" is a noun. Also making the language consistent with the rest of our docs.

`kubeClusterName` sets the name used for the Kubernetes cluster. This name will be shown to Teleport users connecting to the cluster.

## `authentication`

### `authentication.type`

| Type | Default value | Required? | `teleport.yaml` equivalent | Can be used in `custom` mode? |
| - | - | - | - | - |
| `string` | `local` | Yes | `auth_service.authentication.type` | ❌ |

`authentication.type` controls the authentication scheme used by Teleport. Possible values are `local` and `github` for OSS, plus `oidc` and `saml` for Enterprise.

### `authentication.connectorName`

| Type | Default value | Required? | `teleport.yaml` equivalent | Can be used in `custom` mode? |
| - | - | - | - | - |
| `string` | `""` | No | `auth_service.authentication.connector_name` | ❌ |

`authentication.connectorName` sets the default authentication connector.

> See [the SSO documentation](../../enterprise/sso.mdx). Other possible values are `local` and

Reviewer: What are readers supposed to see in the SSO documentation? I would indicate that in this description. 

Since there are only a few SSO connectors, I think it would also make sense to list the possible values here, rather than referring the reader to another docs page.

Author: This field is either the authentication connector name or `local`/`passwordless`. IIUC SSO connector name != supported SSO providers. Users can give any name they want to their SSO connector and create multiple connectors for the same provider. Hence I cannot list all possible values

> [`passwordless`](../../access-controls/guides/passwordless.mdx#optional-enable-passwordless-by-default).

Reviewer: Should the other values have links as well?

Author: Done.

Defaults to `local`.

### `authentication.localAuth`

| Type | Default value | Required? | `teleport.yaml` equivalent | Can be used in `custom` mode? |
| - | - | - | - | - |
| `bool` | `true` | No | `auth_service.authentication.local_auth` | ❌ |

`authentication.localAuth` controls whether local authentication is enabled.
When disabled, users can only log in through authentication connectors like `saml`, `oidc` or `github`.

[Disabling local auth is required for FedRAMP / FIPS](../../enterprise/fedramp.mdx#teleport-auth-server).

### `authentication.lockingMode`

| Type | Default value | Required? | `teleport.yaml` equivalent | Can be used in `custom` mode? |
| - | - | - | - | - |
| `string` | `""` | No | `auth_service.authentication.locking_mode` | ❌ |

`authentication.lockingMode` controls the locking mode cluster-wide.

> See [the locking modes documentation](../../access-controls/guides/locking.mdx#next-steps-locking-modes)

Reviewer: Since there are only two possible values, I think we can list/describe these here, rather than referring the user to another docs page.

Author: Done
